docs/topics/gradle/gradle-compiler-options.md

```diff
@@ -136,6 +136,120 @@ tasks.named('compileKotlin', org.jetbrains.kotlin.gradle.tasks.KotlinCompilation
 </tab>
 </tabs>
 
+### Migrate from `kotlinOptions{}` to `compilerOptions{}` {collapsible="true"}
+
+Before Kotlin 2.2.0, you could configure compiler options using the `kotlinOptions{}` block. Since the `kotlinOptions{}`
+block is deprecated in Kotlin 2.2.0, this section provides guidance and recommendations for migrating your build
+scripts to use the `compilerOptions{}` block instead.
+
+#### Centralize compiler options and use types
+
+Whenever possible, configure compiler options at the [extension level](#extension-level), and override them for specific tasks
+at [compilation unit level](#compilation-unit-level).
+
+To make your configuration safer and more expressive, avoid using raw strings and prefer strongly typed values. For example, if you have:
+
+```kotlin
+plugins {
+    kotlin("jvm") version "2.2.0"
+}
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+    kotlinOptions {
+        jvmTarget = "17"
+        languageVersion = "2.2"
+        apiVersion = "2.2"
+    }
+}
+```
+
+After migration, this becomes:
+
+```kotlin
+plugins {
+    kotlin("jvm") version "2.2.0"
+}
+
+kotlin {
+    // Extension level
+    compilerOptions {
+        jvmTarget = JvmTarget.fromTarget("17")
+        languageVersion = KotlinVersion.fromVersion("2.2")
+        apiVersion = KotlinVersion.fromVersion("2.2")
+    }
+}
+
+// Override at compilation unit level
+tasks.named<KotlinJvmCompile>("compileKotlin"){
+    compilerOptions {
+        apiVersion = KotlinVersion.fromVersion("2.0")
+    }
+}
+```
+
+#### Migrate away from `android.kotlinOptions`
+
+If your build script previously used `android.kotlinOptions`, migrate to `kotlin.compilerOptions` instead. Either at
+the extension level or the target level.
+
+For example, if you have:
+
+```kotlin
+plugins {
+    id("com.android.application")
+    kotlin("android")
+}
+
+android {
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+}
+```
+
+Update it to:
+
+```kotlin
+plugins {
+    id("com.android.application")
+    kotlin("android")
+}
+
+kotlin {
+    androidTarget {
```

Comment:
`androidTarget` is not available in `kotlin-android` plugin. So it should be just extension level example